client/state/sites/reducer.js

```diff
@@ -216,6 +217,20 @@ export function items( state = null, action ) {
 				},
 			};
 		}
+
+		case EDITOR_TYPE_SET: {
+			const { siteId, editor } = action;
+			if ( state[ siteId ] && editor.match( /^(classic|gutenberg)$/ ) ) {
+				return {
+					...state,
+					[ siteId ]: {
+						...state[ siteId ],
+						editor: editor,
+					},
+				};
+			}
+			return state;
+		}
```

Comment:
how about we reuse the existing abstractions?

```js
const editorType = ( state = 'classic', { type, editor } ) =>
	type === EDITOR_TYPE_SET
		? editor
		: state;

export default keyedReducer( 'siteId', editorType );
```

Reply:
Thanks for pointing this out! `sites` has a variety of different ways to handle its reducers, so I just went for the one I grokked first that made sense. I like this more though – except for the fact that I must be missing something, since the reducer don't reduce no more (dispatching the action doesn't update the store value). Do you see it?

Reply:
let's make sure to remove this now that we have the `selectedEditor` subtree